src/_pytest/python_api.py

```diff
@@ -708,12 +717,19 @@ def approx(expected, rel=None, abs=None, nan_ok: bool = False) -> ApproxBase:
         expected = _as_numpy_array(expected)
         cls = ApproxNumpy
     elif (
-        isinstance(expected, Iterable)
+        hasattr(expected, "__getitem__")
```

Comment:
Unfortunately `Sequence` requires many other methods (such as `__reversible__`, `__contains__`), which would break the previous assumption made by `test_generic_ordered_sequence`.